packages/server/src/core/tests.rs

```diff
@@ -413,7 +430,7 @@ async fn test_duplicated_courses() {
 // Test core function in a full flow
 // ------------------------------------------------------------------------------------------------------
 
-async fn get_catalog(catalog: &str) -> Catalog {
+pub async fn get_catalog(catalog: &str) -> Catalog {
```

Comment:
This is dangerous. This function should not be marked public as it is only viable for tests.
You have 2 solutions: 
1. Add `#[cfg(test)]` to the line above to make sure this function only compiles for testing.
2. Remove the `pub` and move catalog_validations testing here.

Generally I prefer solution 2 because I don't see the value of having a whole folder for catalog validations, given we only have a single validation to check for cycles in the `CourseBank` dependency graph.

I would create a `catalog_validations.rs` file in `core` directory, and add the new unit test currently located in `catalog_validations/tests.rs` to this file instead.